packages/drivers/odsp-driver-definitions/src/odspCache.ts

```diff
@@ -127,5 +127,10 @@ export interface IPersistedCache {
  * @internal
  */
 export function getKeyForCacheEntry(entry: ICacheEntry): string {
-	return `${entry.file.docId}_${entry.type}_${entry.key}`;
+	const version =
+		"fileVersion" in entry.file.resolvedUrl && entry.file.resolvedUrl.fileVersion !== undefined
+			? `_${entry.file.resolvedUrl.fileVersion}`
+			: "";
+	const suffix = entry.type === snapshotKey ? "" : `_${entry.key}`;
```

Comment:
FileVersion is already part of entry.key. 

export function createCacheSnapshotKey(
	odspResolvedUrl: IOdspResolvedUrl,
	snapshotWithLoadingGroupId: boolean | undefined,
): ICacheEntry {
	const cacheEntry: ICacheEntry = {
		type: snapshotWithLoadingGroupId ? snapshotWithLoadingGroupIdKey : snapshotKey,
		key: odspResolvedUrl.fileVersion ?? "",
		file: {
			resolvedUrl: odspResolvedUrl,
			docId: odspResolvedUrl.hashedDocumentId,
		},
	};
	return cacheEntry;
}

Reply:
That's the case for snapshots, but not ops - the key for ops is `${this.batchSize}_${batchNumber}`, but the version is contained on the url on the entry (in the case of ODSP driver).